web/server/codechecker_server/api/report_server.py

```diff
@@ -2534,20 +2557,44 @@ def getLinesInSourceFileContents(self, lines_in_files_requested, encoding):
         self.__require_view()
         with DBSession(self._Session) as session:
             res = defaultdict(lambda: defaultdict(str))
-            for lines_in_file in lines_in_files_requested:
-                if lines_in_file.fileId is None:
-                    LOG.warning("File content requested without a fileId.")
-                    LOG.warning(lines_in_file)
-                    continue
-                sourcefile = session.query(File).get(lines_in_file.fileId)
-                cont = session.query(FileContent).get(sourcefile.content_hash)
+
+            contents = []
+            for chunk in util.chunks(
+                    lines_in_files_requested, SQLITE_MAX_VARIABLE_NUMBER):
```

Comment:
Please, add a short comment which tells that the goal of this chunking is not for performance, but for being compatible with SQLite. However, I think it is a relevant question in many other queries where SQLite is used. We should find some general solution for handling SQLite limitations without complicating the code in case of PostgreSQL.

I'm also thinking if we really need to support large data in case of SQLite. We are proposing that DBMS for simplified CodeChecker usage with less amount of data.

Reply:
Added a comment.

I agrree with that, we should find a better way to handle these queries. Maybe with an enchanced ORM/Model layer, with proper methods to work with data could facade these from the API endpoint code.
Regarding your SQLite comment, isn't there a way to configure  SQLite to work with larger variable sizes in query? Maybe enfoce it somehow through CodeChecker. Or maybe give a warning about an unsupported version, which cannot handle large data?